pkg/storageincentives/redistributionstate_test.go

```diff
@@ -0,0 +1,183 @@
+// Copyright 2023 The Swarm Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package storageincentives
+
+import (
+	"context"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethersphere/bee/pkg/log"
+	erc20mock "github.com/ethersphere/bee/pkg/settlement/swap/erc20/mock"
+	"github.com/ethersphere/bee/pkg/statestore/mock"
+	transactionmock "github.com/ethersphere/bee/pkg/transaction/mock"
+	"math/big"
+	"testing"
+)
+
+func createRedistribution(t *testing.T, erc20Opts []erc20mock.Option, txOpts []transactionmock.Option) *RedistributionState {
+	t.Helper()
+	if erc20Opts == nil {
+		erc20Opts = []erc20mock.Option{
+			erc20mock.WithBalanceOfFunc(func(ctx context.Context, address common.Address) (*big.Int, error) {
+				return big.NewInt(1000), nil
+			}),
+		}
+	}
+	if txOpts == nil {
+		txOpts = []transactionmock.Option{
+			transactionmock.WithTransactionFeeFunc(func(ctx context.Context, txHash common.Hash) (*big.Int, error) {
+				return big.NewInt(1000), nil
+			}),
+		}
+	}
+	state, err := NewRedistributionState(log.Noop, common.Address{}, mock.NewStateStore(), erc20mock.New(erc20Opts...), transactionmock.New(txOpts...))
+	if err != nil {
+		t.Fatal("failed to connet")
+	}
+	return state
+}
+func TestState(t *testing.T) {
+	t.Parallel()
+	input := Status{
+		Phase:           commit,
+		IsFrozen:        false,
+		Round:           2,
+		LastWonRound:    2,
+		LastPlayedRound: 2,
+		LastFrozenRound: 2,
+		Block:           2,
+		Fees:            big.NewInt(0),
+		Reward:          big.NewInt(0),
+	}
+	expected := Status{
+		Phase:           commit,
+		IsFrozen:        false,
+		Round:           2,
+		LastWonRound:    2,
+		LastPlayedRound: 2,
+		LastFrozenRound: 0,
+		Block:           2,
+		Fees:            big.NewInt(0),
+		Reward:          big.NewInt(0),
+	}
+	t.Run("all state success", func(t *testing.T) {
+		t.Parallel()
+		state := createRedistribution(t, nil, nil)
+		if len(input.Phase.String()) > 0 && input.Round > 0 && input.Block > 0 {
+			state.SetCurrentEvent(input.Phase, input.Round, input.Block)
+		}
+		state.SetLastWonRound(input.LastWonRound)
+		state.SetFrozen(input.IsFrozen, input.LastFrozenRound)
+		state.SetLastPlayedRound(input.LastPlayedRound)
+		got, err := state.Status()
+		if err != nil {
+			t.Fatal("failed to get state")
+		}
+		if got != nil && (got.IsFrozen != expected.IsFrozen || got.LastFrozenRound != expected.LastFrozenRound ||
```

Comment:
You can use the `cmp.Diff` function for comparison (as in other tests):
```Golang
if diff := cmp.Diff(want, have); diff != "" {
	t.Errorf("Clone(...): result mismatch (-want +have):\n%s", diff)
}
```

Reply:
This doesn't work as we have two `big.Int` properties. it would always fail. 

I have also used these. It won't work.

`reflect.DeepEqual(got, have)`
`cmp.Equal(got, have)`

Reply:
`firstWinResult.Reward.Cmp(expectedReward) != 0` works, right ?

Reply:
> This doesn't work as we have two big.Int properties. it would always fail.

You don't specify why it would fail, but I assume it's because of the unexported fields. If this is the case, you can pass the following option to the `cmp.Diff` call: `cmp.Option{cmp.AllowUnexported(new(big.Int))}`.

Reply:
Done thanks guys.